Samples/auth/Outlook-Add-in-SSO-events/routes/getUserProfile.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const express = require('express');
+var router = express.Router();
+const authHelper = require('../server-helpers/obo-auth-helper');
+const getGraphData = require('../server-helpers/msgraph-helper');
+const jwt = require('jsonwebtoken');
+
+router.get(
+  '/getUserProfile',
+  authHelper.validateJwt,
+  async function (req, res) {
+    try {
+      const authHeader = req.headers.authorization;
+      let oboRequest = {
+        "oboAssertion": authHeader.split(' ')[1],
+        "scopes": ["user.read"],
+      };      
+
+      // The Scope claim tells you what permissions the client application has in the service.
+      // In this case we look for a scope value of access_as_user, or full access to the service as the user.
+      const tokenScopes = jwt.decode(oboRequest.oboAssertion).scp.split(' ');
+      const accessAsUserScope = tokenScopes.find(
+        (scope) => scope === 'access_as_user'
+      );
+      if (!accessAsUserScope) {
+        res.status(401).send({ "type": "Missing access_as_user" });
+        return;
+      }
+      const cca = authHelper.getConfidentialClientApplication();
+      const response = await cca.acquireTokenOnBehalfOf(oboRequest);
+      const rootUrl = '/me';
+      
+      // Note that the last parameter, for queryParamsSegment, is hardcoded. If you reuse this code in
+      // a production add-in and any part of queryParamsSegment comes from user input, be sure that it is
+      // sanitized so that it cannot be used in a Response header injection attack.
```

Comment:
```suggestion
      // a production add-in and any part of queryParamsSegment comes from user input, be sure that it's
      // sanitized so that it can't be used in a Response header injection attack.
```